src/mod/channels.mod/tclchan.c

```diff
@@ -1512,12 +1513,26 @@ static int tcl_channel_modify(Tcl_Interp *irp, struct chanset_t *chan,
       p = strchr(item[i], ':');
       if (p) {
         *p++ = 0;
-        *pthr = atoi(item[i]);
-        *ptime = atoi(p);
-        *--p = ':';
+        if ((!strtol(item[i], &endptr1, 10) && (strcmp(endptr1, "\0") != 0)) \
+           || (!strtol(p, &endptr2, 10) && (strcmp(endptr2, "\0") != 0))) {
+          *--p = ':';
+          if (irp)
+            Tcl_AppendResult(irp, "values must be integers: ", item[i], NULL);
+          return TCL_ERROR;
+        } else {
+          *pthr = atoi(item[i]);
+          *ptime = atoi(p);
+          *--p = ':';
+        }
       } else {
-        *pthr = atoi(item[i]);
-        *ptime = 1;
+        if (!strtol(item[i], &endptr1, 10) && (endptr1[0] != "\0")) {
```

Comment:
Comparing a `char` to a `char[]`? As above (but the inverse, `endptr` must point to the end of the string indicating a valid 0), use `!(*endptr)`.